src/datahike/db.cljc

```diff
@@ -1553,26 +1553,25 @@
         keep-history? (and (-keep-history? db) (not (no-history? db a-ident))
                            (not= :db/txInstant a-ident))
         op-count      (.op-count db)
-        old-val (when keep-history?
-                  (:v (first (di/-slice (:eavt db)
-                                        (dd/datom (.-e datom) (.-a datom) nil (.-tx datom))
-                                        (dd/datom (.-e datom) (.-a datom) nil (.-tx datom))
-                                        :eavt))))]
+        old-datom (first (di/-slice (:eavt db)
```

Comment:
This seems to be much slower than as I had it, at least in the jobtech tests. It also makes having an upsert function unnecessary. So like `old-val` retrieving the `old-datom` should be done in the index namespace.

Reply:
For some reason this also breaks the benchmarks I am trying to run, although that doesn't seem to make much sense to me.